packages/flutter/lib/src/material/drawer.dart

```diff
@@ -353,7 +359,7 @@ class DrawerControllerState extends State<DrawerController> with SingleTickerPro
   }
 
   void _settle(DragEndDetails details) {
-    if (_controller.isDismissed)
+    if (_controller.isDismissed || _controller.isCompleted)
```

Comment:
I believe this is a corner case that https://github.com/flutter/flutter/commit/1d195cb99523dcd2858f1eff76373212c87d57f7#diff-35bed6db55813abd82965aa8f24e568b missed.
This cause that if you drag the drawer all the way to open state, it will fire another automatic forward animation that have zero duration even if it is not required.